frontend-react/src/network/api/NewApi.ts

```diff
@@ -90,27 +84,49 @@ export const buildEndpointUrl = <P extends RSUrlParams>(
     }
 };
 
-/* Handles generating the config from parameters */
-export const createAxiosConfig = <P extends RSUrlParams>(
+/* Case insensitive way of checking if an endpoint contains a method.
+ * example: "GET" === "get" || "GET" === "GET" */
+export const endpointHasMethod = (
+    api: API,
+    endpointKey: string,
+    method: Method
+) => {
+    const endpoint = extractEndpoint(api, endpointKey);
+    const canAccessMethod = endpoint.methods
+        .map((m) => m.toUpperCase())
+        .includes(method.toUpperCase());
+    if (!canAccessMethod)
+        throw Error(`Method ${method} cannot be used by ${endpointKey}`);
+    return canAccessMethod;
+};
+
+/* Handles generating the config from inputs with checks in the middle */
+export const createRequestConfig = <P extends StringIndexed, D = any>(
```

Comment:
[food for thought] I'm not convinced of the need for a request config here. Curious to talk about what this function gives us vs. what it complicates

Reply:
It consolidates. Rather than loosely calling `buildEndpointUrl` and `endpointHasMethod` on our `API` objects throughout the app, or muddying up the endpoint hook with this logic, it guarantees that all we have to do is define an API object, and pass in params to `useEndpoint` inside whatever component we wanna use it. And in the case that it can't construct it because of an invalid input, we have an Error return type. We can check `instanceof SimpleError` to ensure our result is a valid config and then be on our way inside of `useEndpoint`.

Reply:
would also like to get on a call about this